### ISSUE TITLE
Use Fragments instead of Preference class mashup

### ISSUE THREAD

**Issue by mwarning** [mwarning/trigger#19]
This would also remove some hacks.

**Comment by stikonas** [comment]
> parse door return data in various ways (e.g. json)

This would make app useful for other purposes too, e.g. turning lights on/off. Now LOCKED/UNLOCKED is expected in return data...

**Comment by mwarning** [comment]
Yes, this is rather limited right now. But making it more generic might also is a UI problem.
How to device that that specific value is present? Maybe a regex. But that might be rather complex to type.

**Comment by stikonas** [comment]
Regex is probably an overkill. Maybe it's easier to type exact message (similarly to open/close commands).

Regarding UI, that's unfortunately true :(. There are already quite a few fields in that MQTT dialog.

**Comment by mwarning** [comment]
@stikonas I have added two settings that default to LOCKED and UNLOCKED. They can be regular expressions as well. The code is not yet test.

I think it is ok, since the main target are DIY door systems. In use, someone needs to figure out the parameters and then can exchange the setup via QR-Code. :]

**Comment by stikonas** [comment]
> @stikonas I have added two settings that default to LOCKED and UNLOCKED. They can be regular expressions as well. The code is not test yet.
> 
> I think it is ok, since the main target are DIY door systems. In use, someone needs to figure out the parameters and then can exchange the setup via QR-Code. :]

Thanks. It should work. I reviewed the commit, and it looks good to me (I don't really code in Java but it's readable anyway :)).
